netshare/learn/utils/word2vec_embedding.py

```diff
@@ -35,4 +46,99 @@ def word2vec_train(
         model.save(model_path)
     logger.info(f"Word2Vec model is saved at {model_path}")
 
-    return model_path
+    return None
+
+
+def get_word2vec_type_col(word2vec_cols: List[Any]) -> Dict[str, List[str]]:
+    dict_type_cols: Dict[str, List[str]] = {}
+    for col in word2vec_cols:
+        type = col.encoding
+        if type not in dict_type_cols:
+            dict_type_cols[type] = []
+        dict_type_cols[type].append(col.column)
+    return dict_type_cols
+
+
+def build_annoy_dictionary_word2vec(
+    df: pd.DataFrame,
+    model_path: str,
+    word2vec_cols: List[Any],
+    word2vec_size: int,
+    n_trees: int,
+) -> None:
+
+    dict_type_annDictPair = {}
+
+    model = Word2Vec.load(model_path)
+    wv = model.wv
+
+    # type : [cols]
+    # ("ip": ["srcip", "dstip"])
+    # "word2vec_port": ["srcport", "dstport"]
+    # "word2vec_proto": ["proto"]
+    dict_type_cols = get_word2vec_type_col(word2vec_cols)
+    logger.debug(f"word2vec type columns are: {dict_type_cols}")
+
+    for type, cols in dict_type_cols.items():
+        type_set = set(
+            list(itertools.chain.from_iterable([list(df[col]) for col in cols]))
+        )
+        type_ann = AnnoyIndex(word2vec_size, "angular")
+        type_dict = {}
+        index = 0
+
+        for ele in type_set:
+            type_ann.add_item(index, get_vector(model, str(ele), norm_option=True))
+            type_dict[index] = ele
```

Comment:
```suggestion
        for index, ele in enumerate(type_set):
            type_ann.add_item(index, get_vector(model, str(ele), norm_option=True))
            type_dict[index] = ele
```
Syntax sugar :)

Reply:
don't you prefer that `type_dict` will be a list?

Reply:
`type_dict`, is used to help to find out the nearest 'word'. Therefore I don't need to iterate it and expect its correct order. I think maybe dict is better here? WDYT?